.husky/pre-commit

```diff
@@ -0,0 +1,2 @@
+npm run prettier
+git add .
```

Comment:
Really?

Reply:
This really works

Reply:
Lint fix is required as prettier doesn't know how to handle switch statements and a few other rules. The combination of the two produces the best quality code.

Reply:
I was actually referring to the `git add .`, should this really be added?

Reply:
You are right, I've changed it

Reply:
I don't believe that `git add` does anything. In fact, how do you distinguish between changes by prettier and other changes by the developer?

Reply:
`git add` doesn't do anything, I've removed it. Changes made by prettier will be unstaged just like other changes the developer don't want to commit.

Reply:
I believe you'd need to stash all unstaged changes, then run prettier, then stage all changes, then restore stashed changes.

Reply:
@mtrezza That is exactly what happens minus the restore stashed part.

Reply:
This happens with that commands in the hook?

Reply:
Basically copied the previous implementation https://github.com/parse-community/Parse-SDK-JS/blob/alpha/package.json#L132

I wanted to add the linted changes to the commit.

Reply:
>This happens with what commands in the hook?

Sorry I mean if prettier itself fails as in an `unexpected token error` would cause the commit to fail.

Reply:
How would you get a failing lint with this setup of prettier + lint:fix?

Reply:
Got it, it wouldn't fail, but it would restyle the code, right? I personally don't like code being restyled while I'm still working on it. That's just my personal preference. How are we doing this in other repos, do they all have precommit restyling, or is this repo an exception?

Reply:
Parse Server has precommit hooks for styling, the Docs repo has a hook for bundling assets.

Reply:
Well let's align with parse server then.

Reply:
It will happen automatically, I changed it back to be close to the original implementation and align with parse server.